### DISCUSSION TITLE
Optimal symprec value ?

### DISCUSSION THREAD

**Discussion by husakm** [spglib/spglib#440] [Q&A]
Did anybody try to determine optimal symprec value ?
I see you suggest 0.00001
I had found an article benchmarking findsym on 50 000 inorganic structures, suggesting 0.001 .
Material Studio code use 0.1.
I am ready to do a benchmarko on 200 000 structures from Cambridge structure database and simply try different values and see what will happen , but I do not want to rediscover America ...

**Comment by atztogo** [comment]
I think you have to resdicover your America. But if you know how the current spglib works, you may be able to do it systematically.

Crystal symmetries are defined for perfect structure, i.e., no tolerance is allowed as the concept. So we have to define what toleration means mathematically to fit distorted crystal structure into some higher symmetry. In spglib, it is not well defined.

Internally, there are several hard-corded criteria of measure of tolerance in spglib. Those values were determined in a heuristic way so that it matches for my purpose.  Practically, I wanted spglib to work nicely finding some symmetry of distorted crystal structures tolerating ~0.1 angstrom (A), at least I hoped not to crash. You may not believe spglib can crash, but it was indeed very usual to crash with larger tolerance like 0.1 A. In fact, strictly speaking, spglib behaviour is even unit dependent.

After many years, spglib became somehow robust. But due to its algorithm, we may fall into unfortunate cases that the symmetry as a user expects may not be found always. For example, I want to find higher symmetry to lower symmetry with decreasing tolerance value, but it is not guaranteed. I guess this unfortunate case can happen more frequently with a larger tolerance, although I didn't meet such a case in my practical use cases. To reduce unfortunate cases under the current algorithm, we have to make iterative tolerance adjustment more precise, but as a drawback, the computational performance of spglib will be degraded.

We may be able to have a better algorithm, but at least I am satisfied with current implementation. For people who want to improve it, there are plenty of space to develop new algorithms and also improve performance. At the same time, once we make useful software, since it will become popular, resource required for maintenance will be non-negligible, so the balance between required effort for maintenance and better performance against implementation of algorithms should also be considered finally.

**Comment by husakm** [comment]
I need to find a value able to separate experimentaly determined structures with correcly and incorrectly assigned space group. For 0.1 symprec value spglib finds for aprox. 1500 structuers from CSD (containing 1 500 000 structures total) higher spg than the published. Unfortunately there is no glue whatever this is reality or false hits. Only for a few structures the incorrect spg was found by others and confirmed .... I will probably run the check with multiple synprec values and chceck what will hapen . I thing about a generation of "synthetic" structures with artificialy created and known displacement from ideal to see what works ...

**Comment by husakm** [comment]
Maybe you can be interested in following results. I had run spglib on 194017 crystal structures form CSD database (selected hog-quality one form lat 10 years)
Results table:
```
**symprec     spglib_crash(no result)  SG not match experiment**
0.00001             0                                     7809
0.0001              0                                     7117  
0.001               2                                     3133   
0.01               12                                       19  
0.1               137                                      200
```
It sounds like spglib have issues to work with high tolerance ...
It sound like the symprec should be balanced between false negative and to much tolerance
Unfortunately I do not know, whatever the truth is on spglib side or experiment side.
More investigation needed ...

**Reply by LecrisUT** [reply]
Thank you, this is a very valuable data to have. It would be useful to have a formatted structure containing:
- What test structures are used
- What experimental result is reported
- What Spglib dataset reports

I did not study the crystal structure semantics, so I can't give any valuable technical comments, but from what I've seen on other issues, there are crystal structures notations that are equivalent, which should be taken into consideration.

@atztogo what info would help you in this case?

**Reply by atztogo** [reply]
Hall symbols are used as the dataset of matrix representations of symmetry operations to identify the space group type. I think it is good for spglib to concentrate on this job, and different settings may be supported by other codes, since they are separable. I am not at all a specialist on crystallography and am not sure about different settings. I see the Gemmi project does some job, https://gemmi.readthedocs.io/en/latest/symmetry.html.

**Comment by husakm** [comment]
On following links is an excel sheet with the detailed test results:
https://docs.google.com/spreadsheets/d/16woAGPar398zNKegpocbU3cbwiaBTMLH/edit?usp=drive_link&ouid=105101340302149935493&rtpof=true&sd=true
The sheet contains refcodes to the CSD database + origonal and spglib determiend SG. All information about the structures can be than downloaded by the help of the refcode for free from following link:
https://www.ccdc.cam.ac.uk/structures/
I had listed the refcodes only for the problematic structures (not the 194 017 set) in the sheet.
The data are stored in so cold CIF (crystallographic information file) with contains totally all information.. We have an C#/spglib interface able to do the CIF-> spglib data transfer. I believe Avgadro code can do the same work . All structures were transformed to P1 group before Spglib test.  I test SG international number found = this should be unique and difference indicate an issue ... 
I am open to help anybody to replicate such test (I am a scientist teaching crystallography on an Univ developing tools for crystals structure validation).

**Reply by atztogo** [reply]
Thanks @husakm for your data. Yes, at least I want spglib not to crash to the tolerance ~0.1 A. It is very helpful if those crystal structures are provided in spglib python format, and if @husakm could check those crystal structures in the spglib python format are generated as expected for @husakm because I hear CIF parsing is not as easy. Pymatgen may be also tested to compare with @husakm's CIF parser. I think this is the first step.

**Comment by husakm** [comment]
To be honest I am not familiar with python. I have no idea how to transfer the structures from CIF to python spglib format. If you want to investigate the problematic cases (spglib crashes when there is a 0 in the table), it will be more easy for me to generate a C code with the structure data compatible with the Spglib C samples (like corrundum, rutile e.t.c) than to study what a python spglib format looks like . I can generate the data replicating what I do = structures description in P1 inside the unit cell.  I believe there must exist a python CIF parser. We had developed our C# CIF parser our-self (take us 1 year to fine toone it) ...

**Comment by husakm** [comment]
I had put 12 structures in CIF format (problematic in 0.001 and higher range, always generating spglib crash) on a shared disk:
https://drive.google.com/file/d/1PRdnHhfg2gGtWfI2CmkrG6cT1c8ytr4G/view?usp=sharing
If you have a Python CIF parser, you should be able to replicate my results ...

**Comment by atztogo** [comment]
@husakm, without mutual agreement about crystal structure inputs to spglib, we may have to pay unnecessary effort. This happens often in answering spgib questions. Using CIF for communication could also result in unfortunate misunderstanding of problems. So it would be helpful to generate C codes for us not in CIF.

**Reply by LecrisUT** [reply]
I cam try to mediate and extract the info from CIF usimg gemmi. If I don't get to it in 2 weeks, ping me about it. I should have time then at the very least

**Comment by husakm** [comment]
I agree with you. I just work on CIF - > C source code converter. It is a work for a few hours + I will be able to test the C code and issue reproduction on my side ... I hope I will have results today evening ....

**Reply by LecrisUT** [reply]
@husakm, could I ask you to write your replies in the UI of [`https://github.com/spglib/spglib/discussions/440`](https://github.com/spglib/spglib/discussions/440). It is fairly hard to follow the comments, with what is a reply to what.

**Comment by husakm** [comment]
Hi
If you agree I can automatically generate from CIF following C code for the problematic cases.
The code bellow is from DMSO , CSD code DMETSO06. For this structure spglib correctly determine SG as P-1 always. So this is nto a problematic case ...

```c
#include "spglib.h"
#include <stdio.h>
#include <stdlib.h>
#include <math.h>


static void test_spg_get_international(void);

int main(void)
{
	test_spg_get_international();

  return 0;
}

static void test_spg_get_international(void)
{
	//section  generated by checkCIF-DFT
	int num_atom = 20;
	double lattice[3][3] = {
	{5.1729998588562,2.55255341529846,0.743359446525574},
	{0,5.02483510971069,2.07075071334839},
	{0,0,6.5218939781189}
	};

	double position[20][3] = {
	{0.601899981498718,0.164000004529953,0.149900004267693},
	{0.330000013113022,0.120999999344349,0.203600004315376},
	{0.907000005245209,0.792999982833862,0.247999995946884},
	{0.913999974727631,0.686699986457825,0.160999998450279},
	{0.872300028800964,0.706099987030029,0.385399997234344},
	{0.089400053024292,0.800899982452393,0.24549999833107},
	{0.666000008583069,0.284000009298325,0.337000012397766},
	{0.528299987316132,0.475899994373322,0.31580001115799},
	{0.859000027179718,0.268900007009506,0.326400011777878},
	{0.64520001411438,0.173500001430511,0.470400005578995},
	{0.398100018501282,0.835999965667725,0.850099980831146},
	{0.669999957084656,0.879000008106232,0.796400010585785},
	{0.0929999947547913,0.207000002264977,0.751999974250793},
	{0.0860000252723694,0.313300013542175,0.83899998664856},
	{0.127699971199036,0.293900012969971,0.614600002765656},
	{0.910599946975708,0.199100002646446,0.754500031471252},
	{0.333999991416931,0.715999960899353,0.662999987602234},
	{0.471700012683868,0.524100005626678,0.68419998884201},
	{0.140999972820282,0.731099963188171,0.6735999584198},
	{0.35479998588562,0.826499998569489,0.529600024223328},
	};

	int types[20] = {
	16,8,6,1,1,1,6,1,1,1,16,8,6,1,1,1,6,1,1,1,
	};
	//end of section  generated by checkCIF-DFT


  int num_spg;
  char symbol[21];
  double symprec;

  symprec = 1e-5;
  num_spg = spg_get_international(symbol, lattice, position, types, num_atom, symprec);

  printf("*** Example of spg_get_international ***:\n");
  if ( num_spg > 0 ) {
    printf("%s (%d) symprec %f\n", symbol, num_spg, symprec);
  } else {
    printf("Space group could not be found. Symprec = %f\n",symprec);
  }

  symprec = 0.001;
  num_spg = spg_get_international(symbol, lattice, position, types, num_atom, symprec);

  printf("*** Example of spg_get_international ***:\n");
  if (num_spg > 0) {
	  printf("%s (%d) symprec %f\n", symbol, num_spg, symprec);
  }
  else {
	  printf("Space group could not be found. Symprec = %f\n", symprec);
  }
}
```

**Comment by husakm** [comment]
Following is an automatically generated code for structure with CSD CODE DAWROL
The structure should be  C2/m = 12, but it is not granted 
For symprec = 1E-5, spglib gives P2
For symprec 0.01 spglib crashes with no result
Please, let me know if I should prepare for your tests more problematic structures in the C format. I can put them zipped on my google drive + post a link. I will probably prepare only the data part of the C, not the whole source ...

```c
#include "spglib.h"
#include <stdio.h>
#include <stdlib.h>
#include <math.h>


static void test_spg_get_international(void);

int main(void)
{
	test_spg_get_international();

  return 0;
}

static void test_spg_get_international(void)
{
	//section  generated by checkCIF-DFT DAWROL
	int num_atom = 150;
	double lattice[3][3] = {
	{12.0509996414185,-6.22799859684164E-07,-3.76382398605347},
	{0,14.2480001449585,-5.42887789833912E-07},
	{0,0,7.79486799240112}
	};

	double position[150][3] = {
	{0.176029995083809,0.245719999074936,0.248800039291382},
	{0.222800001502037,0.300000011920929,0.265900015830994},
	{0.200680002570152,0.168809995055199,0.172389984130859},
	{0.264499992132187,0.171299993991852,0.139500021934509},
	{0.130879998207092,0.0876199975609779,0.144189953804016},
	{0.147300004959106,0,0.0532000064849854},
	{0.0468299984931946,0,0.853799998760223},
	{0.0917000025510788,0,0.782700002193451},
	{0.966390013694763,0.913969993591309,0.809130012989044},
	{0.987810015678406,0.83652001619339,0.7291100025177},
	{0.0509000010788441,0.83840000629425,0.694700002670288},
	{0.917249977588654,0.757059991359711,0.699899971485138},
	{0.932299971580505,0.70550000667572,0.646000027656555},
	{0.277559995651245,0,0.062999963760376},
	{0.36395999789238,0,0.219230055809021},
	{0.298249989748001,0,0.940720021724701},
	{0.585870027542114,0,0.260859996080399},
	{0.610499978065491,0,0.186199992895126},
	{0.686749994754791,0,0.422399997711182},
	{0.431300014257431,0.995100021362305,0.217700004577637},
	{0.75220000743866,0.0366999991238117,0.418099999427795},
	{0.714600026607513,0.936699986457825,0.454699993133545},
	{0.662699997425079,0.0265999995172024,0.505200028419495},
	{0.823970019817352,0.754279971122742,0.751200020313263},
	{0.777199983596802,0.699999988079071,0.734099984169006},
	{0.799319982528687,0.83118999004364,0.827610015869141},
	{0.735499978065491,0.828700006008148,0.860499978065491},
	{0.869120001792908,0.91237998008728,0.855809986591339},
	{0.852699995040894,0,0.946799993515015},
	{0.953170001506805,0,0.146199941635132},
	{0.90829998254776,0,0.217300057411194},
	{0.033610001206398,0.0860299989581108,0.190870046615601},
	{0.012190000154078,0.163479998707771,0.2708899974823},
	{0.949100017547607,0.16159999370575,0.305299997329712},
	{0.0827500000596046,0.242939993739128,0.300099968910217},
	{0.0676999986171722,0.29449999332428,0.353999972343445},
	{0.722440004348755,0,0.936999976634979},
	{0.636039972305298,0,0.780770003795624},
	{0.701750040054321,0,0.0592800378799438},
	{0.176029995083809,0.754279971122742,0.248800039291382},
	{0.222800001502037,0.699999988079071,0.265900015830994},
	{0.200680002570152,0.83118999004364,0.172389984130859},
	{0.264499992132187,0.828700006008148,0.139500021934509},
	{0.130879998207092,0.91237998008728,0.144189953804016},
	{0.966390013694763,0.0860299989581108,0.809130012989044},
	{0.987810015678406,0.163479998707771,0.7291100025177},
	{0.0509000010788441,0.16159999370575,0.694700002670288},
	{0.917249977588654,0.242939993739128,0.699899971485138},
	{0.932299971580505,0.29449999332428,0.646000027656555},
	{0.823970019817352,0.245719999074936,0.751200020313263},
	{0.777199983596802,0.300000011920929,0.734099984169006},
	{0.799319982528687,0.168809995055199,0.827610015869141},
	{0.735499978065491,0.171299993991852,0.860499978065491},
	{0.869120001792908,0.0876199975609779,0.855809986591339},
	{0.033610001206398,0.913969993591309,0.190870046615601},
	{0.012190000154078,0.83652001619339,0.2708899974823},
	{0.949100017547607,0.83840000629425,0.305299997329712},
	{0.0827500000596046,0.757059991359711,0.300099968910217},
	{0.0676999986171722,0.70550000667572,0.353999972343445},
	{0.414129972457886,0,0.739140033721924},
	{0.389500021934509,0,0.813799977302551},
	{0.313250005245209,0,0.577600002288818},
	{0.568699955940247,0.995100021362305,0.782299995422363},
	{0.24779999256134,0.0366999991238117,0.581900000572205},
	{0.285399973392487,0.936699986457825,0.545300006866455},
	{0.337300002574921,0.0265999995172024,0.494799971580505},
	{0.389500021934509,0.999899983406067,0.813799977302551},
	{0.568699955940247,0.00490000005811453,0.782299995422363},
	{0.24779999256134,0.963299989700317,0.581900000572205},
	{0.285399973392487,0.0632999986410141,0.545300006866455},
	{0.337300002574921,0.973399996757507,0.494799971580505},
	{0.610499978065491,0.999899983406067,0.186199992895126},
	{0.431300014257431,0.00490000005811453,0.217700004577637},
	{0.75220000743866,0.963299989700317,0.418099999427795},
	{0.714600026607513,0.0632999986410141,0.454699993133545},
	{0.662699997425079,0.973399996757507,0.505200028419495},
	{0.676029980182648,0.745720028877258,0.248800039291382},
	{0.722800016403198,0.800000011920929,0.265900015830994},
	{0.700680017471313,0.66881000995636,0.172389984130859},
	{0.764500021934509,0.671299993991852,0.139500021934509},
	{0.630879998207092,0.58762001991272,0.144189953804016},
	{0.647300004959106,0.5,0.0532000064849854},
	{0.546829998493195,0.5,0.853799998760223},
	{0.59170001745224,0.5,0.782700002193451},
	{0.466390013694763,0.413969993591309,0.809130012989044},
	{0.487809985876083,0.33652001619339,0.7291100025177},
	{0.550899982452393,0.33840000629425,0.694700002670288},
	{0.417250007390976,0.257059991359711,0.699899971485138},
	{0.432300001382828,0.20550000667572,0.646000027656555},
	{0.777559995651245,0.5,0.062999963760376},
	{0.863960027694702,0.5,0.219230055809021},
	{0.798249959945679,0.5,0.940720021724701},
	{0.0858700275421143,0.5,0.260859996080399},
	{0.110499978065491,0.500100016593933,0.186199992895126},
	{0.186749935150146,0.5,0.422399997711182},
	{0.931300044059753,0.495099991559982,0.217700004577637},
	{0.25220000743866,0.536700010299683,0.418099999427795},
	{0.214600086212158,0.436699986457825,0.454699993133545},
	{0.162699937820435,0.526600003242493,0.505200028419495},
	{0.323970019817352,0.254280000925064,0.751200020313263},
	{0.277199983596802,0.199999988079071,0.734099984169006},
	{0.299319982528687,0.33118999004364,0.827610015869141},
	{0.235500007867813,0.328700006008148,0.860499978065491},
	{0.369120001792908,0.412380009889603,0.855809986591339},
	{0.352699995040894,0.5,0.946799993515015},
	{0.453170001506805,0.5,0.146199941635132},
	{0.40829998254776,0.5,0.217300057411194},
	{0.533609986305237,0.586030006408691,0.190870046615601},
	{0.512189984321594,0.66347998380661,0.2708899974823},
	{0.449099987745285,0.66159999370575,0.305299997329712},
	{0.582750022411346,0.742940008640289,0.300099968910217},
	{0.567700028419495,0.79449999332428,0.353999972343445},
	{0.222440004348755,0.5,0.936999976634979},
	{0.13604000210762,0.5,0.780770003795624},
	{0.201750010251999,0.5,0.0592800378799438},
	{0.676029980182648,0.254280000925064,0.248800039291382},
	{0.722800016403198,0.199999988079071,0.265900015830994},
	{0.700680017471313,0.33118999004364,0.172389984130859},
	{0.764500021934509,0.328700006008148,0.139500021934509},
	{0.630879998207092,0.412380009889603,0.144189953804016},
	{0.466390013694763,0.586030006408691,0.809130012989044},
	{0.487809985876083,0.66347998380661,0.7291100025177},
	{0.550899982452393,0.66159999370575,0.694700002670288},
	{0.417250007390976,0.742940008640289,0.699899971485138},
	{0.432300001382828,0.79449999332428,0.646000027656555},
	{0.323970019817352,0.745720028877258,0.751200020313263},
	{0.277199983596802,0.800000011920929,0.734099984169006},
	{0.299319982528687,0.66881000995636,0.827610015869141},
	{0.235500007867813,0.671299993991852,0.860499978065491},
	{0.369120001792908,0.58762001991272,0.855809986591339},
	{0.533609986305237,0.413969993591309,0.190870046615601},
	{0.512189984321594,0.33652001619339,0.2708899974823},
	{0.449099987745285,0.33840000629425,0.305299997329712},
	{0.582750022411346,0.257059991359711,0.300099968910217},
	{0.567700028419495,0.20550000667572,0.353999972343445},
	{0.914129972457886,0.5,0.739140033721924},
	{0.889500021934509,0.500100016593933,0.813799977302551},
	{0.813250005245209,0.5,0.577600002288818},
	{0.068699985742569,0.495099991559982,0.782299995422363},
	{0.74779999256134,0.536700010299683,0.581900000572205},
	{0.785399973392487,0.436699986457825,0.545300006866455},
	{0.837300002574921,0.526600003242493,0.494799971580505},
	{0.068699985742569,0.504899978637695,0.782299995422363},
	{0.74779999256134,0.463299989700317,0.581900000572205},
	{0.785399973392487,0.563300013542175,0.545300006866455},
	{0.837300002574921,0.473399996757507,0.494799971580505},
	{0.931300044059753,0.504899978637695,0.217700004577637},
	{0.25220000743866,0.463299989700317,0.418099999427795},
	{0.214600086212158,0.563300013542175,0.454699993133545},
	{0.162699937820435,0.473399996757507,0.505200028419495},
	};

	int types[150] = {
	6,1,6,1,6,6,6,1,6,6,1,6,1,6,8,8,8,1,6,1,1,1,1,6,1,6,1,6,6,6,1,6,6,1,6,1,6,8,8,6,1,6,1,6,6,6,1,6,1,6,1,6,1,6,6,6,1,6,1,8,1,
	6,1,1,1,1,1,1,1,1,1,1,1,1,1,1,6,1,6,1,6,6,6,1,6,6,1,6,1,6,8,8,8,1,6,1,1,1,1,6,1,6,1,6,6,6,1,6,6,1,6,1,6,8,8,6,1,6,1,6,6,6,
	1,6,1,6,1,6,1,6,6,6,1,6,1,8,1,6,1,1,1,1,1,1,1,1,1,1,1,1,
	};
	//end of section  generated by checkCIF-DFT



  int num_spg;
  char symbol[21];
  double symprec;

  symprec = 1e-5;
  num_spg = spg_get_international(symbol, lattice, position, types, num_atom, symprec);

  printf("*** Example of spg_get_international ***:\n");
  if ( num_spg > 0 ) {
    printf("%s (%d) symprec %f\n", symbol, num_spg, symprec);
  } else {
    printf("Space group could not be found. Symprec = %f\n",symprec);
  }

  symprec = 0.01;
  num_spg = spg_get_international(symbol, lattice, position, types, num_atom, symprec);

  printf("*** Example of spg_get_international ***:\n");
  if (num_spg > 0) {
	  printf("%s (%d) symprec %f\n", symbol, num_spg, symprec);
  }
  else {
	  printf("Space group could not be found. Symprec = %f\n", symprec);
  }
}
```

**Reply by LecrisUT** [reply]
Thank you. There are the test data in `test/functional/python/{data,ref}`, and we could use a similar format to process the experimental data. We would need to check these examples, and the licensing of the experimental data, and after that we can make a plan.

**Reply by atztogo** [reply]
Thanks. I will have a look.
@LecrisUT, how can I enable `debug_print`?

**Reply by LecrisUT** [reply]
`cmake -DSPGLIB_DEBUG=ON ...`. It would also be useful to add `-DCMAKE_BUILD_TYPE=Debug` and use the vscode debugger to step through the code.

**Reply by atztogo** [reply]
@husakm, some distances between atoms are closer than the tolerance.

@LecrisUT, @lan496, I expected to get `SPGERR_ATOMS_TOO_CLOSE`, but I could not get it although excecution passes `atoms_too_close:` in `spglib.c`. If this is not my mistake, this is not the expected behaviour. But might this already be reported by @LecrisUT?

**Reply by LecrisUT** [reply]
I would need to run it through the debugger to check the where `SPGERR_ATOMS_TOO_CLOSE` is supposed to be "raised". What issue I have reported before was that depending on the interface, "exception" could be returned or not, but I thought the main interface through `spg_get_database` was catching these issues. This one might be related like that. Can you try it again through `spg_get_database`, etc.?

But if that were the issue, does setting a small enough `symprec` give the appropriate experimental result? I thought in the previous discussion it was unstable even with low `symprec`, but I'm not sure if it was small enough for those though.

@husakm wait a bit until we fix this "exception" raising issue, then could you run again and filter out those issues.

**Reply by husakm** [reply]
Take your time. I can not unfortunately guaranty the structural data are OK - they are taken from the database without any additional check. I was surprised  spglib gives something at low symprec value but crashes for high one (finds no symmetry). This should not happen.

**Reply by husakm** [reply]
According licensing of the experimental data : All data from CSD are free to any experiential use and re-evalaution. They are taken form open scientific literature = nothing secret or proprietary. I have a legal access to the CSD database + I already discus my result with CSD people. There is a chance to find a set of structures for witch a wrong SG determination was already experimentally confirmed. I will try to extract such examples + test what spglib will give.

**Reply by lan496** [reply]
Let's discuss this case in https://github.com/spglib/spglib/issues/445

**Comment by husakm** [comment]
For your inspiration. I had found following article:
"Some more space-group corrections", L.M.Henling, R.E. Marsh, Acta Cryst. (2014), C70, 834-836
https://scripts.iucr.org/cgi-bin/paper?fa3343
The article discuss a 156 incorrect space group determination from about 100 000 CSD entries checked.
I had proceed all of the mentioned 156 structures by SPGLIB.
The best agreement with he article I have got for the largest tested symprec (0.4) :
In 50 cases the incorrect SG detection was reproduced
In 1 case different SG than in the article was found (P c instead of P21/c)
In 16 cases Spglib crashed
In  89 cases Spglib was not able to find the different symmetry and confirmed original incorrect symmetry ...
The article data are freely accessible. I can convert them to anything you want.
Just let me know if you want to do some tests on such real-live examples ...
